lua/neotest-busted/output_handler.lua

```diff
@@ -59,14 +63,23 @@ return function(options)
             duration = handler.getDuration(),
         }
 
-        local ok, result = pcall(json.encode, test_results)
+        local ok, result = pcall(json.encode, test_results, {
+            exception = function(reason, value, state, default_reason)
+                local state_short = table.concat(state.buffer, "")
+                state_short = "..."
+                    .. state_short:sub(#state_short - 100)
+                    .. tostring(state.exception)
+                io.stderr:write(default_reason .. "(" .. state_short .. ")")
+            end,
+        })
 
         if ok then
             file:write(result)
             file:close()
         else
             io_write("Failed to encode test results to json: " .. result .. "\n")
             io_flush()
+            os.exit(1)
```

Comment:
Why is this necessary? Is it because the process cannot continue in a sane way if json encoding failed?

Reply:
Busted is often run as a separate process wrapped in `status, erro = pcal(.._l` for example. Setting the exit code, allows to assert the error.
I have actually updated the json formatter here https://github.com/lunarmodules/busted/pull/748/files

Reply:
I see but couldn't you assert on the error if you returned `false` as the second return value? Or called `error`? I don’t believe `os.exit` is necessary for `pcall`/`xpcall` to register an error.

Nice that you contributed to busted (and that it seems maintained). We can wait and see what the outcome of that PR is since it would probably be a good idea to align with what busted’s builtin handlers do.